File: 프로그래머스/lv0/120924. 다음에 올 숫자/다음에 올 숫자.py
def solution(common):
    flag = True # 등차수열
    tmp = 0
    for i in range(0,len(common)-3+1, 3):
        a, b, c = common[i], common[i+1], common[i+2]
        if a + c ==  2*b:
            flag = True
            tmp = b-a
        elif a*c == b**2:
            flag = False
            tmp = b//a
    
    print(flag, tmp)
    
    if flag:
        return common[-1] + tmp
    else:
        return common[-1]*tmp
            
            

File: 프로그래머스/0/120924. 다음에 올 숫자/다음에 올 숫자.java
class Solution {
    public int solution(int[] common) {
        int a = common[0], b = common[1], c = common[2];
        if (a+c == 2*b) {
            return common[common.length-1] + b-a;
        }
        return common[common.length-1]*(b/a);
    }
}